pkg/operator/ceph/pool/radosnamespace/controller.go

```diff
@@ -261,6 +262,45 @@ func (r *ReconcileCephBlockPoolRadosNamespace) reconcile(request reconcile.Reque
 		return reconcile.Result{}, errors.Wrap(err, "failed to save cluster config")
 	}
 
+	poolAndRadosNamespaceName := fmt.Sprintf("%s/%s", pool, getRadosNamespaceName(cephBlockPoolRadosNamespace))
+	if cephBlockPoolRadosNamespace.Spec.Mirroring.Enabled {
+		err = cephclient.EnableRBDRadosNamespaceMirroring(r.context, r.clusterInfo, poolAndRadosNamespaceName, cephBlockPoolRadosNamespace.Spec.Mirroring.RemoteNamespace, string(cephBlockPoolRadosNamespace.Spec.Mirroring.Mode))
+		if err != nil {
+			return reconcile.Result{}, errors.Wrap(err, "failed to enable rbd rados namespace mirroring")
+		}
+
+		// Schedule snapshots
+		if len(cephBlockPoolRadosNamespace.Spec.Mirroring.SnapshotSchedules) > 0 {
+			err = cephclient.EnableSnapshotSchedules(r.context, r.clusterInfo, poolAndRadosNamespaceName, cephBlockPoolRadosNamespace.Spec.Mirroring.SnapshotSchedules)
+			if err != nil {
+				return reconcile.Result{}, errors.Wrapf(err, "failed to enable snapshot scheduling for rbd rados namespace %q", poolAndRadosNamespaceName)
+			}
+		}
+	} else {
+		mirrorInfo, err := cephclient.GetPoolMirroringInfo(r.context, r.clusterInfo, poolAndRadosNamespaceName)
+		if err != nil {
+			return reconcile.Result{}, errors.Wrapf(err, "failed to get mirroring info for the radosnamespace %q", poolAndRadosNamespaceName)
+		}
+
+		if mirrorInfo.Mode == "image" {
+			mirroredPools, err := cephclient.GetMirroredPoolImages(r.context, r.clusterInfo, poolAndRadosNamespaceName)
+			if err != nil {
+				return reconcile.Result{}, errors.Wrapf(err, "failed to list mirrored images for radosnamespace %q", poolAndRadosNamespaceName)
+			}
+
+			if len(*mirroredPools.Images) > 0 {
+				msg := fmt.Sprintf("there are images in the radosnamespace %q. Please manually disable mirroring for each image", poolAndRadosNamespaceName)
+				logger.Errorf(msg)
+				return reconcile.Result{}, errors.New(msg)
```

Comment:
Why log the error and then return the error? Should we just return it instead?

Reply:
@sp98 added this similar for blockpool I thought it is required.

Reply:
At the top level, the error should be logged, so if you log it here, likely it will be duplicated in the log.